frontend/src/metabase/reference/databases/TableList.jsx

```diff
@@ -100,8 +99,10 @@ class TableList extends Component {
       loading,
     } = this.props;
 
+    const tables = Object.values(entities);
```

Comment:
Apart from fixing the issue, this line here is the main change.
It defines tables as an array in one place and all other functions in this spec now know what to expect.

Using a mix of `Object.keys(entities)` and `Object.values(entities)` randomly was outright confusing.
Not to mention the use of `Object.values(tables)` at https://github.com/metabase/metabase/pull/30751/files#diff-3341dfa456411686e965258000bfbfce2ac7b853cd8d8d96336af0680d3bb9eeL63

🤯